### PR TITLE
Keep improving the test suite

### PR DESCRIPTION
**Pull request by deivid-rodriguez**
I started this to try clear @kaspth's path in #4558. But I added some other improvements in the tests while working on that.

Not a very focused PR, really, but I figure it's small enough to not need splitting.

### CODE REVIEW

File: lib/active_admin/application.rb [hunk 1]
```diff
@@ -178,9 +178,9 @@ def namespace(name)
       name ||= :root
 
       namespace = namespaces[name] ||= begin
-        namespace = Namespace.new(self, name)
-        ActiveSupport::Notifications.publish ActiveAdmin::Namespace::RegisterEvent, namespace
-        namespace
```
Comment (timoschilling) [inline lib/active_admin/application.rb]: 👎

Comment (deivid-rodriguez) [inline lib/active_admin/application.rb]: Commit removed. Apparently my request for better feedback on code reviews didn't succeed... :cry:

Comment (deivid-rodriguez) [inline lib/active_admin/application.rb]: Anyways, I think it's good to keep this change in `lib` (no matter if it's good or bad) out of this PR.

**Comment by deivid-rodriguez** [comment]
Thanks for reviewing Timo!

**Pull request merged**